File: MatriX/samples/csharp/MiniClient/DiscoNode.cs
using System.Windows.Forms;
using Matrix;

namespace MiniClient
{
    class DiscoNode : TreeNode
    {
        public new string Name { get; set; }
        public string Node { get; set; }
        public Jid Jid { get; set; }

        public DiscoNode(string name, string node, Jid jid)
        {
            Name = name;
            Node = node;
            Jid = jid;

            Text = Name + " / " + Node + " / " + Jid;
        }
    }
}

File: MatriX/samples/vb/MiniClient/DiscoNode.vb
Imports System.Windows.Forms
Imports Matrix

Class DiscoNode
    Inherits TreeNode
    Public Shadows Property Name() As String
        Get
            Return m_Name
        End Get
        Set
            m_Name = Value
        End Set
    End Property
    Private Shadows m_Name As String
    Public Property Node() As String
        Get
            Return m_Node
        End Get
        Set
            m_Node = Value
        End Set
    End Property
    Private m_Node As String
    Public Property Jid() As Jid
        Get
            Return m_Jid
        End Get
        Set
            m_Jid = Value
        End Set
    End Property
    Private m_Jid As Jid

    Public Sub New(argName As String, argNode As String, argJid As Jid)
        Name = argName
        Node = argNode
        Jid = argJid

        Text = Name + " / " + Node + " / " + Jid.ToString
    End Sub
End Class